src/step-function.ts

```diff
@@ -0,0 +1,873 @@
+import { Construct } from "constructs";
+import {
+  Arn,
+  ArnFormat,
+  aws_cloudwatch,
+  aws_iam,
+  aws_stepfunctions,
+  Resource,
+  Stack,
+  Token,
+} from "aws-cdk-lib";
+
+import { FunctionDecl, isFunctionDecl } from "./declaration";
+import {
+  ASL,
+  isASL,
+  isMapOrForEach,
+  MapTask,
+  ParallelTask,
+  StateMachine,
+  States,
+  Wait,
+} from "./asl";
+import { isVTL, VTL } from "./vtl";
+import { makeCallable } from "./callable";
+import { CallExpr, isFunctionExpr } from "./expression";
+import { LogOptions } from "aws-cdk-lib/aws-stepfunctions";
+import { AnyFunction, ensureItemOf } from "./util";
+
+export type AnyStepFunction =
+  | ExpressStepFunction<AnyFunction>
+  | StepFunction<AnyFunction>;
+
+export namespace $SFN {
+  export const kind = "SFN";
+  /**
+   * Wait for a specific number of {@link seconds}.
+   *
+   * ```ts
+   * new ExpressStepFunction(this, "F", (seconds: number) => $SFN.waitFor(seconds))
+   * ```
+   *
+   * @see https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-wait-state.html
+   */
+  // @ts-ignore
+  export function waitFor(seconds: number): void;
+
+  export function waitFor(call: CallExpr, context: VTL | ASL): Wait {
+    if (context.kind !== ASL.ContextName) {
+      throw new Error(
+        `$SFN.wait is only available in the ${ASL.ContextName} context`
+      );
+    }
+    const seconds = call.getArgument("seconds")?.expr;
```

Comment:
Do you plan to support when the caller is `any`?

`(<any>$SFN).waitFor()`

Reply:
or rather, why not just use positional arguments?

Reply:
I should use positional arguments!